applications/RANSApplication/python_scripts/formulations/utilities.py

```diff
@@ -0,0 +1,244 @@
+
+    element_name = element_name + element_suffix
+    condition_name = condition_name + condition_suffix
+    return CreateDuplicateModelPart(
+        formulation.GetBaseModelPart(),
+        formulation.GetName() + "_" + element_name + "_" + condition_name,
+        element_name, condition_name, "")
+
+
+def IsBufferInitialized(formulation):
+    return (formulation.GetBaseModelPart().ProcessInfo[Kratos.STEP] + 1 >=
+            formulation.GetMinimumBufferSize())
+
+
+def InitializePeriodicConditions(base_model_part, model_part, variables_list):
+    properties = model_part.CreateNewProperties(
+        model_part.NumberOfProperties() + 1)
+    pcu = KratosCFD.PeriodicConditionUtilities(
+        model_part, model_part.ProcessInfo[Kratos.DOMAIN_SIZE])
+    for variable in variables_list:
+        pcu.AddPeriodicVariable(properties, variable)
+
+    index = model_part.NumberOfConditions()
+    for condition in base_model_part.Conditions:
+        if condition.Is(Kratos.PERIODIC):
+            index += 1
+            node_id_list = [node.Id for node in condition.GetNodes()]
+            periodic_condition = model_part.CreateNewCondition(
+                "PeriodicCondition", index, node_id_list, properties)
+            periodic_condition.Set(Kratos.PERIODIC)
+
+
+def CreateSteadyScalarScheme(relaxation_factor):
+    return steady_scalar_scheme(relaxation_factor)
+
+
+def CreateSteadyAlgebraicFluxCorrectedScheme(relaxation_factor,
+                                                     boundary_flags,
+                                                     is_periodic):
+    if (is_periodic):
+        return afc_steady_scalar_scheme(relaxation_factor, boundary_flags,
+                                        KratosCFD.PATCH_INDEX)
+    else:
+        return afc_steady_scalar_scheme(relaxation_factor, boundary_flags)
+
+
+def CreateBossakRelaxationScalarScheme(bossak_value, relaxation_factor, scalar_variable,
+                             scalar_rate_variable,
+                             relaxed_scalar_rate_variable):
+    return bossak_scheme(bossak_value, relaxation_factor, scalar_variable,
+                         scalar_rate_variable, relaxed_scalar_rate_variable)
+
+
+def GetBoundaryFlags(boundary_flags_parameters):
+    if (boundary_flags_parameters.size == 0):
+        raise Exception("No boundary flags were found")
+
+    flags = Kratos.KratosGlobals.GetFlag(
+        boundary_flags_parameters[0].GetString())
+    for i in range(1, boundary_flags_parameters.size()):
+        flags |= Kratos.KratosGlobals.GetFlag(
+            boundary_flags_parameters[i].GetString())
+
+    return (flags)
+
+
+def GetDefaultConditionName(model_part):
+    process_info = model_part.ProcessInfo
+    if (process_info.Has(Kratos.DOMAIN_SIZE)):
+        domain_size = process_info[Kratos.DOMAIN_SIZE]
+        if (domain_size == 2):
+            return "LineCondition"
+        elif (domain_size == 3):
+            return "SurfaceCondition"
```

Comment:
This precludes the use of quads and hexas.

Reply:
At the moment RANS only supports 2D triangles and 3D tets because VMS monolithic and Fractional step only supports them.